ui/src/app/views/cohort-edit/component.spec.ts

```diff
@@ -0,0 +1,262 @@
+import {TestBed, async, fakeAsync, tick, ComponentFixture} from '@angular/core/testing';
+import {Title, By} from '@angular/platform-browser';
+import {FormsModule} from '@angular/forms';
+import {ActivatedRoute, UrlSegment} from '@angular/router';
+import {CohortEditComponent} from 'app/views/cohort-edit/component';
+import {RouterTestingModule} from '@angular/router/testing';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+import {SignInService} from 'app/services/sign-in.service';
+import {WorkspaceComponent} from 'app/views/workspace/component';
+
+import {Cohort, CohortListResponse} from 'generated';
+import {CohortsService} from 'generated';
+import {Observable} from 'rxjs/Observable';
+import {Observer} from 'rxjs/Observer';
+
+// TODO: Replace with Swagger-generated model
+class Workspace {
+  id: string;
+  namespace: string;
+  cohorts: Cohort[];
+}
+
+class Context {
+  fixture: ComponentFixture<CohortEditComponent>;
+  route: UrlSegment[];
+  cohortsService: CohortsService;
+  nameField: DebugElement;
+  descriptionField: DebugElement;
+  addButton: DebugElement;
+  saveButton: DebugElement;
+
+  constructor(testBed: typeof TestBed) {
+    this.fixture = testBed.createComponent(CohortEditComponent);
+    this.route = this.fixture.debugElement.injector.get(ActivatedRoute).snapshot.url;
+    this.cohortsService = this.fixture.debugElement.injector.get(CohortsService);
+    this.nameField = this.fixture.debugElement.query(By.css('#name'));
+    this.descriptionField = this.fixture.debugElement.query(By.css('#description'));
+  }
+}
+
+function simulateInput(fixture: ComponentFixture<Component>,
+  element: DebugElement,
+  text: string) {
+  element.nativeNode.value = text;
+  element.nativeNode.dispatchEvent(new Event('input'));
+  updateAndTick(fixture);
+}
+
+function updateAndTick(fixture: any) {
+  fixture.detectChanges();
+  tick();
+}
+
+class CohortsServiceStub {
+  constructor() {
+    const stubWorkspace = new Workspace();
+    stubWorkspace.id = WorkspaceComponent.DEFAULT_WORKSPACE_ID;
+    stubWorkspace.namespace = WorkspaceComponent.DEFAULT_WORKSPACE_NS;
+
+    const exampleCohort: Cohort = {id: '',
+      name: '', description: '', criteria: '', type: ''};
+    exampleCohort.id = '1';
+    exampleCohort.name = 'sample name';
+    exampleCohort.description = 'sample description';
+
+    stubWorkspace.cohorts = [exampleCohort];
+    this.workspaces = [stubWorkspace];
+  }
+  public workspaces: Workspace[];
+
+  private getMatchingWorkspaceOrSendError(wsNamespace: string,
+    wsId: string,
+    observer: Observer<{}>): Workspace {
+    const workspaceFound = this.workspaces.find(function(workspace: Workspace) {
+      if (workspace.namespace === wsNamespace && workspace.id === wsId) {
+        return true;
+      } else {
+        return false;
+      }
+    });
+    if (workspaceFound === undefined) {
+      observer.error(`Error Searching. No workspace ${wsNamespace}, ${wsId} found \
```

Comment:
How about splitting into two interpolated strings to wrap?
```
observer.error(
    `Error Searching. No workspace ${wsNamespace}, ${wsId} found`
    + 'in cohort service stub.');
```
(Or actually just one interpolated + one regular, since there are no vars after the break.)